### PR TITLE
Evaluate rows on addEmptyIndex

### PR DESCRIPTION
**Pull request by JonoPrest**
Fixes edge case where values already exist in the inMemory store when the lookup by index gets run.

### CODE REVIEW

**Review by DZakh — APPROVED** [review]
Nice

File: scenarios/test_codegen/test/LoadLayer_test.res [hunk 1]
```diff
@@ -367,12 +367,7 @@ describe("LoadLayer", () => {
...
-    // FIXME: Should register indexes for entities set in the inMemoryStore before the index creation
-    // Assert.deepEqual(users, [user1])
-    // Assert.deepEqual(mock.loadEntitiesByIdsCalls, [])
-    // Assert.deepEqual(mock.loadEntitiesByFieldCalls, [])
-
-    Assert.deepEqual(users, [])
+    Assert.deepEqual(users, [user1])
     Assert.deepEqual(mock.loadEntitiesByIdsCalls, [])
     Assert.deepEqual(
       mock.loadEntitiesByFieldCalls,
```
[the remark this answers is not in the document]
Comment (JonoPrest) [reply scenarios/test_codegen/test/LoadLayer_test.res:373]: Yes if it's the first time adding the index 👍🏼
